fetchcode/api.py

```diff
@@ -26,29 +27,45 @@ class Response:
 - `content_type`: content type of the file
 - `size`: size of the retrieved content in bytes
 - `url`: fetched URL
+- `scheme` : scheme of the URL
     """
-    def __init__(self, location, content_type, size, url):
+    def __init__(self, location, content_type, size, url, scheme):
         self.location = location
         self.content_type = content_type
         self.size = size
         self.url = url
- 
+        self.scheme = scheme
+
 
-def fetch(url):
+def fetch(url, location=None):
     """
     Return a `Response` object built from fetching the content at the `url` URL string.
     """
-    r = requests.get(url)
 
-    temp = tempfile.NamedTemporaryFile(delete=False)
-    filename = temp.name
+    url_parts = urlparse(url)
+
+    scheme = url_parts.scheme
+
```

Comment:
Why the blank line? Does it help with readability here?

Reply:
Okay got this one,